chain/chain/src/runtime/mod.rs

```diff
@@ -736,8 +739,20 @@ impl RuntimeAdapter for NightshadeRuntime {
         // Total amount of gas burnt for converting transactions towards receipts.
         let mut total_gas_burnt = 0;
         let mut total_size = 0u64;
-        // TODO: Update gas limit for transactions
-        let transactions_gas_limit = gas_limit / 2;
+
+        let transactions_gas_limit =
+            if ProtocolFeature::CongestionControl.protocol_version() <= protocol_version {
```

Comment:
Hm, I would be happy to use a simple function but I don't like to superficially make code more complex. If we use `checked_feature!` with "stable", we use a ~50 lines macro that reduces to a one-liner anyway. And unfortunately, Rust macros are not the most accessible code for devs.
Also I really don't like the implication that the feature has already been stabilized, which it hasn't.

Maybe it's time to add simple function on ProtocolFeature if the raw comparison is not preferred.

Reply:
I don't mind either, I'm fine with sticking to just comparison. 

>  Also I really don't like the implication that the feature has already been stabilized, which it hasn't. 

I don't think this is the implication, checked_feature is merely a check if the feature is stable or not.

Reply:
> > Also I really don't like the implication that the feature has already been stabilized, which it hasn't.
> 
> I don't think this is the implication, checked_feature is merely a check if the feature is stable or not.

Maybe that's not how people see it today. But the macro is written such that the first argument is a string which is "stable" if the feature was stabilized and "some_feature" otherwise. If the latter is the case, the macro uses conditional compilation to remove code based on compile time flags. Otherwise, code is always included because the feature is stable.

It's hard for me to see how the string "stable" could imply anything other than that the feature is stable ^^

But I guess it doesn't make much sense to talk about the semantics of a macro designed purely to make it easier to deal with conditional compilation, if we avoid conditional compilation entirely now.